src/java/org/apache/cassandra/service/accord/serializers/IVersionedWithKeysSerializer.java

```diff
@@ -194,135 +224,80 @@ private static long encodeBitmap(AbstractRanges serialize, AbstractRanges supers
         }
 
         @DontInline
-        private <K extends RoutableKey> void serializeLargeSubset(AbstractKeys<K> serialize, int serializeCount, AbstractKeys<K> superset, int supersetCount, DataOutputPlus out) throws IOException
+        private <K extends Routable, R extends Routables<K>> void serializeLargeSubset(R serialize, int serializeCount,
```

Comment:
having the code copy/pasted all over was hard to deal with.  I moved to having 1 method to do everything for all 3 types rather than special casing all over.